src/ingest_validation_tools/schema_loader.py

```diff
@@ -289,44 +364,51 @@ def get_table_schema(
 
 
 def get_directory_schema(directory_type: str, schema_version: str) -> Optional[dict]:
-    directory_schema_path = _directory_schemas_path / \
-        _get_schema_filename(directory_type, schema_version)
+    directory_schema_path = _directory_schemas_path / _get_schema_filename(
+        directory_type, schema_version
+    )
+    # Roll backward to latest directory schema version if numbers don't match
+    # This does not account for the directory schema number being higher...
     if not directory_schema_path.exists():
-        return None
+        new_schema_version = int(schema_version) - 1
+        if new_schema_version >= 0:
```

Comment:
This will check back to one version before, not sure how this would work if we went from v0 to v2 for example

Reply:
Oh my hope was that it would go like this (but this needs more testing and I am historically not great at recursion):

**[loop 1, looking for "whatever" directory schema v2]**
get_directory_schema("whatever", "2"):
     ....
     if not directory_schema_path.exists():
          new_schema_version = 1
          if new_schema_version >= 0:
               **[loop 2, v2 not found, looking for "whatever" directory schema v1]**
               get_directory_schema("whatever", "1")
                              ....
                              if not directory_schema_path.exists():
                              new_schema_version = 0
                              if new_schema_version >= 0:
                                   **[loop 3, v1 not found, looking for "whatever" directory schema v0]**
                                   get_directory_schema("whatever", "0")
                                        ...
                                        return schema
                    return schema
     return schema

And then if v0 didn't exist...there's obviously a problem.

Whether this is a valid idea based on how schema versions for metadata vs. directories compares....well...we will find out.

Reply:
You are definitely right, I didn't realize that this was recursive when reviewing!